packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx

```diff
@@ -361,17 +366,70 @@ const NewServiceDriverEditor = observer(() => {
     } else {
       newServiceDriver.setMappingOption(val);
     }
+    //reset runtime
+    newServiceDriver.setRuntimeOption({
+      label: runtimeSelectorPlaceholder,
+      value: undefined,
+    });
+  };
+  // runtime
+  const customRuntimeLabel = '(custom)';
+  const availableRuntimes =
+    newServiceDriver.mappingOption !== undefined
+      ? editorStore.graphManagerState.usableRuntimes.filter((runtime) =>
+          runtime.runtimeValue.mappings
+            .map((m) => m.value)
+            .includes(
+              guaranteeNonNullable(newServiceDriver.mappingOption).value,
+            ),
+        )
+      : [];
+  const runtimeOptions =
+    availableRuntimes.length > 0
+      ? availableRuntimes.map(buildElementOption)
+      : [
```

Comment:
May be we can have `custom` runtime in the list always

Reply:
I think we should have custom in the list always as well. 
If there is a compatible runtime then select that but custom should be allowed in the dropdown

Reply:
https://github.com/finos/legend-studio/assets/73408381/dca5a134-545f-4f44-b549-764e1856b428